src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -615,7 +615,14 @@ public Throwable getLastError() {
 			return this.ensureConnectTask(timeout)
 				.thenCompose(webSocket -> {
 					return sendAsyncLock.acquireThenCompose(timeout, () -> {
-						String json = command.getResponse().toJsonString();
+						String json = null;
+						if (command.getResponse() != null) {
+							json = command.getResponse().toJsonString();
+						}
+						if (command.getRenewToken() != null) {
```

Comment:
>if [](start = 6, length = 2)

should this be an `else if` instead? What if none of the if statements match? (All good reasons to encapsulate this inside ListenerCommand.toJsonString(), possibly making ListenerCommand into an interface with toJsonString method).